tests/core/middleware/test_eth_tester_middleware.py

```diff
@@ -20,3 +27,157 @@ def test_get_block_formatters(w3):
     assert keys_diff.pop() == "mixHash"  # mixHash is not implemented in eth-tester
+
+
+sample_address_1 = "0x0000000000000000000000000000000000000001"
+sample_address_2 = "0x0000000000000000000000000000000000000002"
+
+
+@pytest.mark.parametrize(
+    "w3_accounts, w3_coinbase, method, from_field_added, from_field_value",
+    (
+        (sample_address_1, sample_address_2, "eth_call", True, sample_address_2),
+        (
+            sample_address_1,
+            sample_address_2,
+            "eth_estimateGas",
+            True,
+            sample_address_2,
+        ),
+        (
+            sample_address_1,
+            sample_address_2,
+            "eth_sendTransaction",
+            True,
+            sample_address_2,
+        ),
+        (sample_address_1, sample_address_2, "eth_gasPrice", False, None),
+        (sample_address_1, sample_address_2, "eth_blockNumber", False, None),
+        (sample_address_1, sample_address_2, "meow", False, None),
+        (sample_address_1, None, "eth_call", True, sample_address_1),
+        (sample_address_1, None, "eth_estimateGas", True, sample_address_1),
+        (sample_address_1, None, "eth_sendTransaction", True, sample_address_1),
+        (sample_address_1, None, "eth_gasPrice", False, None),
+        (sample_address_1, None, "eth_blockNumber", False, None),
+        (sample_address_1, None, "meow", False, None),
+        (None, sample_address_2, "eth_call", True, sample_address_2),
+        (None, sample_address_2, "eth_estimateGas", True, sample_address_2),
+        (None, sample_address_2, "eth_sendTransaction", True, sample_address_2),
+        (None, sample_address_2, "eth_gasPrice", False, sample_address_2),
+        (None, sample_address_2, "eth_blockNumber", False, sample_address_2),
+        (None, sample_address_2, "meow", False, sample_address_2),
+        (None, None, "eth_call", True, None),
+        (None, None, "eth_estimateGas", True, None),
+        (None, None, "eth_sendTransaction", True, None),
+        (None, None, "eth_gasPrice", False, None),
+        (None, None, "eth_blockNumber", False, None),
+        (None, None, "meow", False, None),
+    ),
+)
+def test_default_transaction_fields_middleware(
+    w3_accounts, w3_coinbase, method, from_field_added, from_field_value
+):
+    w3_accounts = [w3_accounts]
+
+    def mock_request(_method, params):
+        return params
+
+    mock_w3 = Mock()
+    mock_w3.eth.accounts = w3_accounts
+    mock_w3.eth.coinbase = w3_coinbase
+
+    middleware = default_transaction_fields_middleware(mock_request, mock_w3)
+    base_params = {"chainId": 5}
+    filled_transaction = middleware(method, [base_params])
+
+    filled_params = filled_transaction[0]
+
+    assert ("from" in filled_params.keys()) == from_field_added
+    if "from" in filled_params.keys():
+        assert filled_params["from"] == from_field_value
+
+    filled_transaction[0].pop("from", None)
+    assert filled_transaction[0] == base_params
+
+
```

Comment:
nit: I've been separating async from sync tests with:

```py
def test_last_sync_test_here():
    ...
    

# -- async -- #


@pytest.mark.asyncio
async def test_async_first_async_test_here():
    ...
```

With sync and async tests that exist in the same file I'd like to organize them somehow for quicker / easier readability. Since that convention has been used in quite a few places already I'm partial to it 😄 .

Reply:
check